### ISSUE TITLE
Pass Date Parameters as Arguments for SubField

### ISSUE THREAD

**Issue by dharmeshtailor** [graphql-dotnet/graphql-dotnet#803]
We have a graphql query, that has parameters at subfield level.

```
{
  root {
    rootElement1
    rootElement2
    rootElement3
    settlements(startDate: "2018-08-27T00:00:00Z", endDate: "2018-09-02T00:00:00Z") {
      someElement1
      someElement2
      someElement3
    }
  }
}
```

On the server side, arguments within "SomeElement" subfield are parsed as string, although the graphiql shows the datatype of arguments as "Date".

![image](https://user-images.githubusercontent.com/5554788/44826430-53eae980-ac52-11e8-9b2f-db7fe7bfa2be.png)

Although similar approach works perfectly, if the arguments are at root level node.

```
{
  root (startDate: "2018-08-27T00:00:00Z", endDate: "2018-09-02T00:00:00Z"){
    rootElement1
    rootElement2
    rootElement3
    settlements {
      someElement1
      someElement2
      someElement3
    }
  }
}
```

Also passing dates to settlements using query variable works perfectly.

When I debugged through our code, it turns out that context.Arguments is of type Dictionary<string, object>, while the property "Arguments" on type "Field" is of type "Arguments". This is not stopping us in any way, as we are using query variables to pass the dates. But curious to understand why is it this way? 

Code from CoreToVanillaConverter in GraphQL Source to generate field:

![image](https://user-images.githubusercontent.com/5554788/44827127-0fad1880-ac55-11e8-8938-bdd9d31b69ef.png)

Code from ExecutionHelper to generate arguments for ResolveFieldContext.

![image](https://user-images.githubusercontent.com/5554788/44827149-3c613000-ac55-11e8-9ceb-faa6bf70d455.png)

**Comment by joemcbride** [comment]
Can you provide some more context for your issue?

What version are you using?  What server technology are you using (ASP.NET Core MVC, Web API, etc.)?

The Field you’re showing is the object model for the Query. We call that the Abstract Syntax Tree (AST) type.  The Arguments on the AST represent the parsed object model, with the values represented as an IValue.  `CoreToVanillaConverter` exists a bit for historical reasons, though that is converting the GraphQL Parser AST to the AST that the GraphQL .NET framework understands.

https://github.com/graphql-dotnet/graphql-dotnet/blob/bfb23d2653f60f9a5d03a1ab6c134cd417caca69/src/GraphQL/Language/AST/Argument.cs#L19

`ResolveFieldContext.Arguments` represent the computed arguments for the field that are provided to your resolver function.  Those computed values could be from inline values as shown above, variables, or default values.

So in short - one is the AST, one is the computed values.  Related, but two different things.

As far as having trouble passing dates - can you provide a failing test that demonstrates this issue?

**Comment by dharmeshtailor** [comment]
Sorry for missing out on the information:

GraphQL version : 2.0.0-alpha-805
Technology : ASP.Net Web API on Azure Service fabric.

I got that, one is AST and other computed value from root level arguments. but my point was to check, why 2 different ways of representing the arguments?

How do you want me to provide the test?

**Comment by denisedelbando** [comment]
I thnk my issue is similar #828 

https://github.com/graphql-dotnet/graphql-dotnet/issues/828

**Comment by sungam3r** [comment]
@dharmeshtailor Is this issue still actual for you?

**Comment by dharmeshtailor** [comment]
> @dharmeshtailor Is this issue still actual for you?

Not issue anymore.
